src/lib/templateengine.js

```diff
@@ -1407,6 +1407,26 @@ function TemplateEngine( undefined ) {
     }
   };
   
+  /**
+   * Little helper to find the relevant page path for a given widget.
+   * @param element The XML element
+   * @param widgetpath The path / ID of the widget
+   * @return The path of the parent
+   */
+  this.getPageIdForWidgetId = function( element, widgetpath )
+  {
+    var
+      parent = element.parentNode,
+      parentpath = widgetpath.replace( /[0-9]*$/, '' );
+    
+    while( parent.nodeName === 'group' )
```

Comment:
As there are other elements that can contain widget-elements like navbars, infoaction-plugin I guess.`while (parent && parent.nodeName !== 'page')` would be better, right?